### PR TITLE
support .cargo/config.toml

### LINKED ISSUE #9: Support `.cargo/config.toml`
As of <https://github.com/rust-lang/cargo/pull/7295>, Cargo allows using `.cargo/config.toml` instead of just `.cargo/config`. The current recommendation is [to use the former](https://github.com/rust-lang/cargo/blob/rust-1.46.0/CHANGELOG.md#changed-1).  However, `cargo-project` does not currently support `.cargo/config.toml`.

### PR DESCRIPTION
**Pull request by benmkw**
~should be ok~ ;) 

closes https://github.com/japaric/cargo-project/issues/9
ref https://github.com/probe-rs/cargo-embed/issues/29

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -72,12 +72,14 @@ impl Project {
...
         let manifest = parse::<Manifest>(&toml)?;
 
-        // parse .cargo/config
+        // parse .cargo/config or .cargo/config.toml
         let mut target = None;
         let mut target_dir = env::var_os("CARGO_TARGET_DIR").map(PathBuf::from);
-        if let Some(path) = search(root, &cargo_config) {
+        if let Some(path) = search(root, &cargo_config).or_else(|| search(root, &cargo_config_toml))
+        {
             let config: Config = parse(&path.join(&cargo_config))?;
```
Comment (libreamartists) [inline src/lib.rs:82]: If the `or_else` is triggered above, this join should have input `&cargo_config_toml` instead of `&cargo_config`

Comment (benmkw) [reply src/lib.rs:82]: ha thanks! good catch 👍

File: src/lib.rs [hunk 2]
```diff
@@ -72,14 +72,22 @@ impl Project {
         // parse Cargo.toml
         let toml = root.join("Cargo.toml");
         let cargo_config = Path::new(".cargo").join("config");
+        let cargo_config_toml = Path::new(".cargo").join("config.toml");
```
Comment (jonas-schievink) [inline src/lib.rs:75]: This might be a bit clearer:

```suggestion
        let cargo_config_toml = cargo_config.with_extension("toml");
```

**Review by jonas-schievink — APPROVED** [review]

**Comment by japaric** [comment]
sorry, missed this PR but support for this was added in release 0.2.6

**Pull request closed without merging**